test/cmdlineTests/~name_dependent_cse_bug/cse_bug.yul

```diff
@@ -0,0 +1,30 @@
+object "C" {
```

Comment:
Did you check if this yields a failure without the above fix? (I'm sure you did, buy I have to ask).

Reply:
If not, I double-checked it now, and it does generate a different assembly code using solidity version `0.8.21`. This PR indeed fixes that.

```diff
16d15
<                     let mid := var
19c18
<                     default { sstore(a, mid) }
---
>                     default { sstore(a, var) }
48,49c47
<       dup1
<       dup4
---
>       dup3
53d50
<       pop
59,62c56
<       swap3
<       pop
<       dup3
<       swap1
---
>       swap2
63a58
>       dup2
```

Reply:
Yeah I did. But you have a point - it's not very easy to tell without running the code. And the cause is so finicky that it's hard to be sure that this will continue to be a true repro for this issue in the future.

Hard do to anything about it though. The best I could come up with is to add an optimizer test that shows the generated IR. You can check in the diff how it changes with the fix.